### PR TITLE
Merge islamic formatting keys into one

### PR DESCRIPTION
**Pull request by Manishearth**
They're identical, and we shouldn't carry extra data. If they start differing in the future we can introduce more keys.

Part of https://github.com/unicode-org/icu4x/issues/3854

### CODE REVIEW

**Review by sffc — APPROVED** [review]

File: provider/datagen/src/transform/cldr/datetime/mod.rs [hunk 1]
```diff
@@ -346,45 +346,15 @@ impl_data_provider!(
 );
 impl_data_provider!(IndianDateSymbolsV1Marker, symbols::convert_dates, "indian");
 impl_data_provider!(
-    IslamicCivilDateLengthsV1Marker,
+    IslamicDateLengthsV1Marker,
     |dates, _| DateLengthsV1::from(dates),
     "islamicc"
```
Comment (robertbastian) [inline provider/datagen/src/transform/cldr/datetime/mod.rs:351]: What is this used for?

Comment (Manishearth) [reply provider/datagen/src/transform/cldr/datetime/mod.rs:351]: The source cldr calendar

Comment (robertbastian) [reply provider/datagen/src/transform/cldr/datetime/mod.rs:351]: So CLDR does have different data for the different calendars?

Comment (robertbastian) [reply provider/datagen/src/transform/cldr/datetime/mod.rs:351]: *separate data

Comment (Manishearth) [reply provider/datagen/src/transform/cldr/datetime/mod.rs:351]: Yes but it's identical (and should be identical), we can potentially check it.

Comment (robertbastian) [reply provider/datagen/src/transform/cldr/datetime/mod.rs:351]: That's what I'm trying to get at. If it doesn't matter which CLDR calendar we use, we should verify that they're all the same.

**Pull request merged**